cpp/src/arrow/compute/exec/hash_join_node.cc

```diff
@@ -453,6 +453,20 @@ Status HashJoinSchema::CollectFilterColumns(std::vector<FieldRef>& left_filter,
   return Status::OK();
 }
 
+Status ValidateHashJoinNodeOptions(const HashJoinNodeOptions& join_options) {
+  if (join_options.key_cmp.size() < 1 || join_options.left_keys.size() < 1 ||
+      join_options.right_keys.size() < 1) {
+    return Status::Invalid("key_cmp and keys cannot be empty");
+  }
+
+  if ((join_options.key_cmp.size() != join_options.left_keys.size()) ||
+      (join_options.key_cmp.size() != join_options.right_keys.size()) {
+    return Status::Invalid("key_cmp and keys must have the same size");
+  }
```

Comment:
In addition, the left keys and right keys must have length equal to each other.  E.g. `left_keys.size() == right_keys.size()`

Reply:
This is implied in the logic of this if statement. If `key_cmp.size() == left_keys.size()` and `key_cmp.size() == right_keys.size()`, then `left_keys.size() == right_keys.size()` must be true as well.